src/components/ai-input.tsx

```diff
@@ -73,12 +103,27 @@ export const AIInput = ( {
 			// Allow Shift + Enter to create a new line
 			return;
 		} else {
+			setIsTyping( true );
 			handleKeyDown( e );
 		}
 	};
 
+	const handleKeyUpWrapper = () => {
+		if ( typingTimeout ) {
+			clearTimeout( typingTimeout );
+		}
```

Comment:
We'd need to also clear the timeout when the component is mounted. Here's an example of how we could implement it:

```patch
diff --git forkSrcPrefix/src/components/ai-input.tsx forkDstPrefix/src/components/ai-input.tsx
index e1c72d39396247958235be5b024f7f4d08b2e4f4..7e764647216310db93f938f05dc47e8c25427f95 100644
--- forkSrcPrefix/src/components/ai-input.tsx
+++ forkDstPrefix/src/components/ai-input.tsx
@@ -28,9 +28,9 @@ export const AIInput = ( {
 	isAssistantThinking,
 }: AIInputProps ) => {
 	const inputRef = useRef< HTMLTextAreaElement >( null );
+	const typingTimeout = useRef< NodeJS.Timeout >();
 
 	const [ isTyping, setIsTyping ] = useState( false );
-	const [ typingTimeout, setTypingTimeout ] = useState< NodeJS.Timeout | null >( null );
 
 	const { RiveComponent, inactiveInput, thinkingInput, typingInput, startStateMachine } =
 		useAiIcon();
@@ -65,6 +65,15 @@ export const AIInput = ( {
 		typingInput,
 	] );
 
+	useEffect(
+		() => () => {
+			if ( typingTimeout.current ) {
+				clearTimeout( typingTimeout.current );
+			}
+		},
+		[]
+	);
+
 	const handleInput = ( e: React.ChangeEvent< HTMLTextAreaElement > ) => {
 		setInput( e.target.value );
 
@@ -109,15 +118,13 @@ export const AIInput = ( {
 	};
 
 	const handleKeyUpWrapper = () => {
-		if ( typingTimeout ) {
-			clearTimeout( typingTimeout );
+		if ( typingTimeout.current ) {
+			clearTimeout( typingTimeout.current );
 		}
 
-		setTypingTimeout(
-			setTimeout( () => {
-				setIsTyping( false );
-			}, 500 )
-		);
+		typingTimeout.current = setTimeout( () => {
+			setIsTyping( false );
+		}, 500 );
 	};
 
 	const getPlaceholderText = () => {

```

Reply:
Thanks for the suggestion! Addressed in [2a85e9f](https://github.com/Automattic/studio/pull/280/commits/2a85e9f4c85d8fb8e7ce43097395327dcba8ba7a)